tests/e2e/etcd_config_test.go

```diff
@@ -735,3 +735,57 @@ func TestV2DeprecationEnforceDefaultValue(t *testing.T) {
 		})
 	}
 }
+
+func TestEtcdAdvertiseClientUnix(t *testing.T) {
+	e2e.SkipInShortMode(t)
+
+	// Create a temporary directory for the data directory
+	dataDir := t.TempDir()
+	unixSocket := "unix:///tmp/etcd-client.sock"
+
+	// Clean up the socket file after test
+	defer os.Remove("/tmp/etcd-client.sock")
```

Comment:
Could we change this to use a `t.TempDir()`, so we don't need to clean it up and ensure it's unique?

```suggestion
	socketDir := t.TempDir()
	unixSocket := fmt.Sprintf("unix://%s/etcd-client.sock", socketDir)
```